.gitmodules

```diff
@@ -1,6 +1,6 @@
 [submodule "core"]
 	path = core
-	url = git@github.com:NYPL-Simplified/server_core.git
+	url = https://github.com/NYPL-Simplified/server_core.git
```

Comment:
readthedocs could not install the submodules through ssh so switched to https.

Reply:
This is going to cause problems, but I don't remember what the problems are. I do remember we tried doing this once or twice and switched back. Here are my guesses as to where the problems lie:

* This might screw up the Puppet deployment used by NYPL. Possibly also the system for building the Docker images.
* This might make it more difficult to do normal development, since your commits and branches on core won't be associated with your SSH key.

@aslagle Do you remember this at all?

Reply:
I found it: it's a Puppet problem: https://github.com/NYPL-Simplified/circulation/commit/bf7c9ab726ca9792b6c2c091d558ab51f467061c#diff-8903239df476d7401cf9e76af0252622

Our Puppet setup hasn't changed much since 2015, so it's probably still a live problem, but I'll check.

Reply:
Yeah, I think this will cause problems. You might need to give it git credentials so it can install through ssh.

Reply:
Ho-Ling said the problem is that the Puppet server is on an old OS that doesn't support TLS 1.2. So SSH will work but HTTPS won't.

However, _HTTP_ would also work, and all of this code is public, so security of the code isn't an issue. I ran an experiment: if you clone a GitHub repo via HTTP, and then try to push a new branch, the push will go over HTTPS. So we could make these into HTTP submodules without leaking our Github credentials (although _keeping track_ of the credentials is a pain, at least on Linux -- I had to [build something from source](https://askubuntu.com/questions/773455/what-is-the-correct-way-to-use-git-with-gnome-keyring-and-https-repos).)

Reply:
That won't mess up local development?

Reply:
I don't think so; worst case we'll all have to take an extra step to make Git cache our Github credentials locally.

Reply:
This is a problem because readthedocs (.org) is meant for open source projects and giving credentials to install using ssh doesn't seem to be possible. We could use the .com version which is meant for private projects but the .org version is more desirable for us.

Is there a plan on updating the Puppet server? The other alternative I can think of is finding if there is a way to prevent the git submodules to install but we will then not have documentation for `core` .

Reply:
Brett knows more about the Puppet situation, and he's out right now. He and Ho-Ling are looking at options but haven't decided on anything.

I think the best options are:

* Change the .gitmodule file to `https` for the general public, and then change it back to `ssh` in the branches we use for Puppet deployments.
* Change the .gitmodule file to `http` and we all configure the [git credential helper](https://help.github.com/en/articles/caching-your-github-password-in-git) so we don't have to keep typing our password.

Reply:
I prefer the first option but from experience, it is hard to keep track of and easy to override a file with different configurations in different branches. In this case, it seems we'll end up having to use the git credential with either option and it's something else to add to a developer's local workflow.

Reply:
Can we agree to try it with HTTP submodules, with a comment explaining the constraints (NYPL Puppet can't handle HTTPS or SSH, Read The Docs can't handle SSH)?

Reply:
I wouldn't be opposed to this but I'm trying to figure out if there's a different way to solve this. Just to update something similar to this, ssh also doesn't work on github pages but https does.